sdk/helper/keysutil/policy.go

```diff
@@ -2136,17 +2161,26 @@ func (p *Policy) ImportPrivateKeyForVersion(ctx context.Context, storage logical
 		ecdsaKey := parsedPrivateKey.(*ecdsa.PrivateKey)
 		pemBlock, _ := pem.Decode([]byte(keyEntry.FormattedPublicKey))
 		publicKey, err := x509.ParsePKIXPublicKey(pemBlock.Bytes)
-		if err != nil {
+		if err != nil || publicKey == nil {
```

Comment:
Interesting, is it possible for the parse function not returning an error but setting the key as nil? I'm just trying to understand why is the second condition needed.

Reply:
Nah, this condition isn't needed, I think I added it when I was debugging and failed to remove it.

The issue was the second one (need to compare an `*ecdsa.PublicKey` to each other; without the pointer they don't cast correctly and thus return `false` when it should be the same key.